Headers/aermass_container_mod.F90

```diff
@@ -0,0 +1,482 @@
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !MODULE: aermass_container_mod.F90
+!
+! !DESCRIPTION: Module AERMASS\_CONTAINER\_MOD contains the derived type used
+!  to store aerosol data used for computing optical depths and diagnostics.
+!\\
+!\\
+! !INTERFACE:
+!
+MODULE AerMass_Container_Mod
+!
+! USES:
+!
+  USE ErrCode_Mod
+  USE Precision_Mod
+
+  IMPLICIT NONE
+  PRIVATE
+!
+! !PUBLIC MEMBER FUNCTIONS:
+!
+  PUBLIC :: Init_AerMass_Container
+  PUBLIC :: Cleanup_AerMass_Container
+!
+! !PUBLIC DATA MEMBERS:
+!
+  !=========================================================================
+  ! Derived type for aerosol data container
+  !=========================================================================
+  TYPE, PUBLIC :: AerMassContainer
+
+     ! Arrays
+     !========================================================================
+     ! OCFPOA      : OM/OC for POA                      [unitless] - used in carbon_mod
+     ! OCFOPOA     : OM/OC for OPOA, OCPI, OCPO         [unitless] - used in carbon_mod
+     ! BCPI        : Hydrophilic black carbon aerosol   [kg/m3]
+     ! BCPO        : Hydrophobic black carbon aerosol   [kg/m3]
+     ! OCPI        : Hydrophilic organic carbon aerosol [kg/m3]
+     ! OCPO        : Hydrophobic organic carbon aerosol [kg/m3]
+     ! OCPISOA     : Hydrophilic OC + SOA aerosol       [kg/m3]
+     ! SALA        : Accumulation mode seasalt aerosol  [kg/m3]
+     ! ACL         : Accumulation mode Cl aerosol       [kg/m3]
+     ! SALC        : Coarse mode seasalt aerosol        [kg/m3]
+     ! SO4_NH4_NIT : Lumped SO4-NH4-NIT aerosol         [kg/m3]
+     ! SO4         : Sulfate aerosol                    [kg/m3]
+     ! HMS         : Hydroxymethane sulfonate aerosol   [kg/m3]
+     ! NH4         : Ammonium aerosol                   [kg/m3]
+     ! NIT         : Inorganic nitrate aerosol          [kg/m3]
+     ! SLA         : Stratospheric liquid aerosol       [kg/m3]
+     ! SPA         : Stratospheric particulate aerosol  [kg/m3]
+     ! TSOA        : Terpene SOA                        [kg/m3]
+     ! ASOA        : Aromatic + IVOC SOA                [kg/m3]
+     ! OPOA        : Aerosol product of SVOC oxidation  [kg/m3]
+     ! SOAGX       : SOA product of GLYX                [kg/m3]
+     ! SOAIE       : SOA product of IEPOX & HMML        [kg/m3]
+     ! PM25        : Particulate matter < 2.5 um        [kg/m3]
+     ! PM10        : Particulate matter < 10 um        [kg/m3]
+     ! ISOAAQ      : Isoprene SOA (aqueous formation)   [kg/m3]
+     ! SOAS        : Simple SOA                         [kg/m3]
+     ! FRAC_SNA    :
+     ! DAERSL      : Mass of hydrophobic aerosol (Mian Chin)
+     ! WAERSL      : Mass of hydrophilic aerosol (Mian Chin)
+     !========================================================================
+     REAL(fp), ALLOCATABLE :: OCFPOA     (:,:)
```

Comment:
Shouldn't these be POINTER instead of allocatable?  That would allow you to point to them without having to declare these with the TARGET attribute.

Reply:
I updated the code to use pointer instead. I didn't need to declare them as TARGET to point to them, and I'm not sure why. Do you know?

Reply:
You can always point to a POINTER but an allocatable array needs to have a TARGET attribute.